### ISSUE TITLE
Compile error when using DefaultHttpHeaderBindingConfiguration

### ISSUE THREAD

**Issue by ArryMatt** [JamesRandall/FunctionMonkey#62]
Using the following is giving me a compile error:

```
.DefaultHttpHeaderBindingConfiguration(new HeaderBindingConfiguration
{
    PropertyFromHeaderMappings = new Dictionary<string, string>
    {
        {"DogId", "x-dog-id" }
    }
})
```

The error is:

> Unhandled Exception: FunctionMonkey.ConfigurationException: CS1513:} expected SourceFile([8064..8065))CS0119:'Nullable' is a type, which is not valid in the given context SourceFile([8026..8041))CS0103:The name 'lt' does not exist in the current context SourceFile([8042..8044))CS0201:Only assignment, call, increment, decrement, await, and new object expressions can be used as a statement SourceFile([8026..8044))CS0119:'DateTime' is a type, which is not valid in the given context SourceFile([8045..8060))CS0103:The name 'gt' does not exist in the current context SourceFile([8061..8063))CS0103:The name 'TryParse' does not exist in the current context SourceFile([8065..8073))
>    at FunctionMonkey.Compiler.Implementation.AssemblyCompiler.CompileAssembly(IReadOnlyCollection`1 syntaxTrees, IReadOnlyCollection`1 externalAssemblyLocations, OpenApiOutputModel openApiOutputModel, String outputBinaryFolder, String outputAssemblyName, String assemblyNamespace, TargetEnum target)
>    at FunctionMonkey.Compiler.Implementation.FunctionCompiler.Compile() in C:\wip\functionMonkey\Source\FunctionMonkey.Compiler\Implementation\FunctionCompiler.cs:line 86
>    at FunctionMonkey.Compiler.Program.Main(String[] args) in C:\wip\functionMonkey\Source\FunctionMonkey.Compiler\Program.cs:line 47

Using the following on each HttpFunction works just fine:
`.Options(o => o.AddHeaderMapping(x => x.DogId, "x-dog-id"))`

TIA

**Comment by ArryMatt** [comment]
This is a slight red herring.  It happens if the command has a nullable date.  So, doing it on each command individually still errors, if that command has the nullable date.

The default just applied it to all commands and one had a nullable date.

**Comment by JamesRandall** [comment]
Thanks for the update!

I'll take a look at it in the context of nullable value types.

**Comment by JamesRandall** [comment]
Fixed this in v3.0.x which is underway at the moment.

**Comment by JamesRandall** [comment]
Addressed in v3.0.4-beta1. Thanks!
